aiohttp/helpers.py

```diff
@@ -47,6 +47,10 @@ def __new__(cls, login, password='', encoding='latin1'):
         if password is None:
             raise ValueError('None is not allowed as password value')
 
+        if ':' in login:
+            raise ValueError(
+                'A ":" is not allowed in login (RFC 1945#section-11.1)')
```

Comment:
That's strange, because without percent-quoting you're not able to use non-ascii names and passwords what is quite awkward today. And since you actually can have them, quoting colon character doesn't breaks the parser while it's quoted.